### ISSUE TITLE
EKS UserData passes bootstrap.sh without a new line

### ISSUE THREAD

**Issue by dkeightley** [rancher/rancher#25779]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):** Create an EKS cluster, the bootstrap.sh script in UserData is passed with additional arguments (`../cfn-signal`) due to a missing new line (\n).

**Result:** `cfn-signal` is not run, and the bootstrap.sh script receives more arguments than intended.

**Other details that may be helpful:** Also remove the \n from the end of `cfn-signal` command, being the last command this is not needed.

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.3.5

**Comment by daxmc99** [comment]
This is available to test in Rancher master and in the 2.4 release branch

**Comment by izaac** [comment]
Validated this issue in
Rancher version: `v2.4-head (04/09/2020)` commit id: 8439d8bde
Rancher version: `master (04/09/2020)` commit id: 03947c351

Reproduced the issue in:
Rancher version: `v2.4.2`.

The node template gets the UserData with the default values from rancher as (notice the missing `\n` between `${BootstrapArguments}/opt/aws/bin/cfn-signal `) :

```
          UserData: !Base64
            'Fn::Sub': "#!/bin/bash\nset -o xtrace\n/etc/eks/bootstrap.sh ${ClusterName} ${BootstrapArguments}/opt/aws/bin/cfn-signal --exit-code $? --stack  ${AWS::StackName} --resource NodeGroup --region ${AWS::Region}"
```

Rancher `master (04/09/2020)` notice the `\n` between `${BootstrapArguments}\n/opt/aws/bin/cfn-signal`

```
          UserData: !Base64
            'Fn::Sub': "#!/bin/bash\nset -o xtrace\n/etc/eks/bootstrap.sh ${ClusterName} ${BootstrapArguments}\n/opt/aws/bin/cfn-signal --exit-code $? --stack  ${AWS::StackName} --resource NodeGroup --region ${AWS::Region}"
```

Same new line as in master can be noticed in `v2.4-head (04/09/2020)`  but I also edited the UserData from Rancher while creating the EKS cluster just for validation (`\necho $?` at the end of the string):

```
          UserData: !Base64
            'Fn::Sub': "#!/bin/bash\nset -o xtrace\n/etc/eks/bootstrap.sh ${ClusterName} ${BootstrapArguments}\n/opt/aws/bin/cfn-signal --exit-code $? --stack  ${AWS::StackName} --resource NodeGroup --region ${AWS::Region}\necho $?"
```

This fix is also noticeable from the UI:

**v2.4.2**
![Screen Shot 2020-04-09 at 1 04 56 PM](https://user-images.githubusercontent.com/27659/78942183-fcd40a00-7a6d-11ea-83df-a5f9b857701e.png)

**v2.4-head**
![Screen Shot 2020-04-09 at 1 05 20 PM](https://user-images.githubusercontent.com/27659/78942225-107f7080-7a6e-11ea-84da-912334f9de8f.png)
